### ISSUE TITLE
Upgrade: Just upgraded from Joomla 3.10 to Beta2a error

### ISSUE THREAD

**Issue by davidspring** [joomlahenk/fabrik#565]
**Before reporting an issue please check that it has not already been fixed by reviewing the changelog in the Wiki or other open issues that may be pending confirmation. Thank you.**

**What happend**
I downloaded the beta2a. Then turned off the Fabrik component and plugins and updated to Joomla 4.2.5. I then updated the website from PHP 7.4 to 8.1. I then installed beta 2a. I then enabled the fabrik component and the needed plugins. 
 But there is an error in System, Maintenance, Database: 
Error:
You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ''#__fabrik_ratings WHERE field = 'listid' AND UPPER(type) REGEXP '^INT([(][0-...' at line 1
Table 'database1.v9mv2_fabrik_packages' doesn't exist
Select Fabrik. Then click Update Structure but. It did not fix the problem. 

**To Reproduce**

see steps above

**Expected behavior**

no database error. 

**Actual behavior**

database error

**Error message**

You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ''#__fabrik_ratings WHERE field = 'listid' AND UPPER(type) REGEXP '^INT([(][0-...' at line 1
Table 'database1.v9mv2_fabrik_packages' doesn't exist

**Screenshots**

**Additional context**
This is a test website. I will not update my real Fabrik websites until I get this test website working with no errors.

**Comment by trob** [comment]
The rating element is not part of Fabrik4.
Please disable it.

**Comment by davidspring** [comment]
I could not find an element plugin called rating. I did disable the element plugin called thumbs and also the form plugin called comments. I then went to System, Maintenance and selected the Fabrik Administrator Component and clicked Update Structure. Unfortunately the error remained. Is there someplace I should be looking for the fabrik rating element? I have never used this element and do not need it.

**Comment by trob** [comment]
In Fabrik3 there's an element rating which adds a table #__fabrik_ratings.
This is not included in Fabrik4 (at least not now).
It seems you had installed it at some point in your Fabrik3.

Check in System/Extensions

**Comment by davidspring** [comment]
Very strange. I have looked in System Manage Extensions and in System Manage Plugins. I have even gone into the file manager to look for it. But no luck. The name of the test website is ourcommunitynetwork.org. I would be happy to give you admin access. But I am pretty certain that there is no Fabrik plugin called Ratings. But I am willing to keep looking if you can give me more hints as to where it might be. Perhaps I need to go into the database and delete the table???

**Comment by davidspring** [comment]
I also have another nearly identical website that has Fabrik and is still Joomla 3. It does not have a Fabrik extension or plugin called Rating either.

**Comment by davidspring** [comment]
I have also checked the database for this website and could not find anything there. I understand the error message indicates that the error is near '#__fabrik_ratings WHERE field = 'listid' . But in researching this error, I think the real problem may be this: Table 'ctc10d_db1.v9mv2_fabrik_packages' doesn't exist

**Comment by trob** [comment]
Ok.
Do you have a table #__fabrik_ratings in your DB?
Do you have a folder plugins\fabrik_element\rating?
Do you have old update SQL files in administrator\components\com_fabrik\sql\updates\mysql?

**Comment by davidspring** [comment]
Do you have a table #__fabrik_ratings in your DB?
No. There are 16 fabrik tables listed in the database. None of them are named fabrik-ratings. There is a table called fabrik_elements. The fourth column in this table is called plugin which shows the type of plugin used for each element. A careful review of all of these elements confirms that none of them are the rating plugin. 

Do you have a folder plugins\fabrik_element\rating?
I have a folder plugins/fabrik_element with about 40 plugins in it. But there is no folder called rating in the fabrik_elements folder. The only folder beginning with r is called radio button. 

Do you have old update SQL files in administrator\components\com_fabrik\sql\updates\mysql?
Yes. There are 8 sql files here with numbers ranging from 3.0.1 to 4.0. 

I would be happy to give you admin access to my website file manager and database. Alternately, I am willing to continue answering any more questions you may have as I really need to find a solution to this problem.

**Comment by davidspring** [comment]
Making a little progress. I installed the new rating plugin from this issue: 
https://github.com/joomlahenk/fabrik/issues/546
This rating plugin is disabled. But it did not solve the problem.
I then install Fabrik 4 version 2b from issue 546. 
This new version of Fabrik 4 did fix a problem with the Fabrik Form Comment Plugin - leaving me with only one issue left: 
You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ''#__fabrik_ratings WHERE field = 'listid' AND UPPER(type) REGEXP '^INT([(][0-...' at line 1
Table 'ctc10d_db1.v9mv2_fabrik_packages' doesn't exist
Warning: There are tables not up to date!

**Comment by joomlahenk** [comment]
I have the same issue. The problem here is that although you don't have the ratings plugin installed nor have a ratings table, all *.sql files in updates\mysql of F3 will run in sequence after you upgraded to J!4.
You can try to remove all these *.sql and add an empty 3.10.sql file in F3 updates\mysql BEFORE you do the upgrade to J!4

**Comment by skurvish** [comment]
@davidspring If you can give us admin access, and ssh access if possible, I can take a look at this for you. You have my contact on discord so yo can private message to me if you like.

**Comment by joomlahenk** [comment]
I see that all F3 sql files in  updates\mysql are still there after upgrade and install of F4. Those should have been deleted as on line 46 in the install script  com_fabrik.manifest.class.php. Only 4.0.sql should be there.
If you remove all 3*.sql files the problem is solved.

**Comment by skurvish** [comment]
@joomlahenk Your original change in the manifest prefilght only deleted one specific sql update file, it was the one for the component and version 3.6.1 I did not change that. You do have a comment in there that it would probably be a good idea to delete them all, but the code is not there to do so.

**Comment by joomlahenk** [comment]
Yes all 3*.sql must be removed. Only 4.0.sql must be there. My intention was to remove ALL *.sql before we install F4, but I guess I made a typo here. Sorry, I didn't test very well.

**Comment by skurvish** [comment]
OK, I will take care of it. @davidspring Let me fix this and I will get you a component zip to install that will remove them all. Some time later today.

**Comment by skurvish** [comment]
@davidspring upon further investigation there are only about 10 files that need to be deleted and they all exist in administrator/components/con_fabrik/sql/updates/mysql so please manually delete these and try your upgrade again.

We will remove these as part of the install script going forward.

**Comment by skurvish** [comment]
@davidspring I have a version of the component that will delete these files if you wish to try it. I am attaching it here. I have tested it on one of my systems and it seems to work, but could use a second confirmation. Thanks.
[com_fabrik_4.0beta2b.zip](https://github.com/joomlahenk/fabrik/files/10098324/com_fabrik_4.0beta2b.zip)

**Comment by davidspring** [comment]
Great news. I ran the new version beta2b.zip and now the database error is gone. In addition, I checked the file manager administrator\components\com_fabrik\sql\updates\mysql folder and the only sql file there was 4.0. I now have a very easy path to update all of my real Fabrik 3 websites to Fabrik 4 and Joomla 4. Thank you so much for your help! I will be working on updating all of my websites today and tomorrow. I will report here if there are any problems. But I am pretty certain that all of the problems are solved!
